gallery_dl/extractor/postmill.py

```diff
@@ -0,0 +1,204 @@
+            match.group(5) or "").items() if self.acceptable_query(key)}
+
+    def items(self):
+        url = self.root + self.base + self.sorting_path
+        if self.query:
+            url += "?" + urllib.parse.urlencode(self.query)
+
+        while url:
+            response = self.request(url)
+            if response.history:
+                redirect_url = response.url
+                if redirect_url == self.root + "/login":
+                    raise exception.StopExtraction(
+                        "HTTP redirect to login page (%s)", redirect_url)
+
+            for nav in text.extract_iter(response.text,
+                                         '<nav class="submission__nav">',
+                                         '</nav>'):
+                post_url = text.unescape(text.extr(nav, '<a href="', '"'))
+                yield Message.Queue, text.urljoin(url, post_url), \
+                    {"_extractor": PostmillPostExtractor}
+
+            url = text.unescape(text.extr(response.text,
+                                          '<link rel="next" href="', '">'))
+
+    def acceptable_query(self, key):
+        return key in self.whitelisted_parameters or key == "t" or \
+            (key.startswith("next[") and key.endswith("]"))
+
+
+BASE_PATTERN = PostmillExtractor.update({
+    "raddle": {
+        "root"   : None,
+        "pattern": (r"(?:raddle\.me|"
+                    r"c32zjeghcp5tj3kb72pltz56piei66drc63vkhn5yixiyk4cmerrjtid"
+                    r"\.onion)"),
+    }
+})
+SORTING_RE = r"(/(?:hot|new|active|top|controversial|most_commented))?"
+QUERY_RE = r"(?:\?([^#]+))?"
+
+
+class PostmillPostExtractor(PostmillExtractor):
+    """Extractor for a single submission URL"""
+    subcategory = "post"
+    pattern = BASE_PATTERN + r"/f/([\w\d_]+)/(\d+)(?:/.+)?$"
+    example = "https://raddle.me/f/FORUM/123/TITLE"
+
+    def __init__(self, match):
+        PostmillExtractor.__init__(self, match)
+        self.forum = match.group(3)
+        self.post_id = match.group(4)
+
+    def post_urls(self):
+        return (self.root + "/f/" + self.forum + "/" + self.post_id,)
+
+
+class PostmillShortURLExtractor(PostmillExtractor):
+    """Extractor for short submission URLs"""
+    subcategory = "shorturl"
+    pattern = BASE_PATTERN + r"/(\d+)$"
+    example = "https://raddle.me/123"
+
+    def __init__(self, match):
+        PostmillExtractor.__init__(self, match)
+        self.post_id = match.group(3)
+
+    def items(self):
+        url = self.root + "/" + self.post_id
+        response = self.request(url, method="HEAD", allow_redirects=False)
+        full_url = text.urljoin(url, response.headers["Location"])
+        yield Message.Queue, full_url, {"_extractor": PostmillPostExtractor}
+
+
+class PostmillHomeExtractor(PostmillSubmissionsExtractor):
+    """Extractor for the home page"""
+    subcategory = "home"
+    pattern = BASE_PATTERN + r"(/(?:featured|subscribed|all)?)" + SORTING_RE \
+        + QUERY_RE + "$"
```

Comment:
Maybe combining `QUERY_RE` and `$` into one would be useful, as they seem to always be used together ... Same for `SORTING_RE`, now that I look at it.

`QUERY_RE` should be `QUERY_RE + "$"`
`SORTING_RE` should be `SORTING_RE` + `QUERY_RE` + `$`
to minimize `+` operations for patterns.